server-core/engine/src/main/java/io/infectnet/server/engine/core/world/MapImpl.java

```diff
@@ -0,0 +1,89 @@
+package io.infectnet.server.engine.core.world;
+
+import io.infectnet.server.engine.core.entity.Entity;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class MapImpl implements Map {
+  /**
+   * The two-dimensional array to hold all tiles in the {@link Map}
+   */
+  private Tile[][] tiles;
+
+  /**
+   * The HashMap containing all Entities which are on the Map,
+   * with the Entity as the key, and a position as value.
+   */
+  private HashMap<Entity,Tile> entityPositionMap;
+
+  /**
+   * The height of the map, the number pf positions available on the y-axis.
+   */
+  private final int height;
+
+  /**
+   * The width of the map, the number pf positions available on the x-axis.
+   */
+  private final int width;
+
+  /**
+   * Creates a new Map in a size defined by the parameters. All its tiles are generated at random.
+   * @param height limitation of the number of tiles on the y-axis
+   * @param width limitation of the number of tiles on x-axis
+   */
+  public MapImpl(int height, int width) {
+    this.height = height;
+    this.width = width;
+
+    tiles = new Tile[height][width];
+    entityPositionMap = new HashMap<>();
+
+    generateNewMap();
+  }
+
+  @Override
+  public List<Entity> listOfEntitiesVisible(Entity entity) {
+    return null;
+  }
+
+  /**
+   * Generates a new array of Tiles with a Cellular Automaton.
+   */
+  private void generateNewMap() {
+    CellularAutomaton cellularAutomaton = new CellularAutomaton(height, width);
```

Comment:
`CellularAutomaton` should not be used directly. Instead it should be provided as a `MapGeneratorStrategy` constructor parameter.